### PR TITLE
fix: respect reconnect=false and clamp server-supplied retry: 0

### PR DESCRIPTION
**Pull request by kinyoklion**
## Summary

Stacked on top of #134. Addresses three pre-existing reconnect-control gaps surfaced during the multi-agent review of that PR.

1. **`retry: 0` collapses backoff to zero.** A server emitting `retry: 0` set `BackoffRetry::base_delay` to `Duration::ZERO`, after which every reconnect path (`client.rs:474, 525, 547, 612` and the new parse-error path) computed `next_delay() == 0` and reconnected immediately — a tight loop. Clamps `change_base_delay` to a 1 ms floor.

2. **EOF arm ignored `reconnect_opts.reconnect`.** The body-exhausted branch unconditionally scheduled `WaitingToReconnect` even when reconnect was disabled. Now honors the flag and transitions to `StreamClosed`, matching every other error path.

3. **Parse-error arm with `reconnect=false` left the parser poisoned.** No state transition happened, so the next poll drained the broken body to EOF — where (2) above papered over the bug. Now transitions to `StreamClosed` so the documented \"do not use the stream after error\" contract holds.

## Context

Stacked PR — base is the `rl/sdk-2345/parser-error-reconnect-state` branch from #134, **not** main. Will retarget once #134 lands.

Tracked in [SDK-2347](https://launchdarkly.atlassian.net/browse/SDK-2347). Predecessor: [SDK-2345](https://launchdarkly.atlassian.net/browse/SDK-2345) / #134.

Surfaced from the multi-agent review of #134 (findings 1, 2, and the corresponding suggested follow-ups 4 and 5). All three were pre-existing — #134 only made the parse-error path more visible.

## Test plan

- [x] `test_change_base_delay_clamps_to_minimum` (`retry.rs`) — pins the 1 ms floor against `change_base_delay(Duration::ZERO)` and a sub-floor value.
- [x] `parser_error_closes_stream_when_reconnect_disabled` (`client.rs`) — asserts the stream emits one `InvalidLine` then `None` when reconnect is off.
- [x] `eof_closes_stream_when_reconnect_disabled` (`client.rs`) — asserts the stream emits the event, `Eof`, then `None` when reconnect is off.
- [x] Existing `parser_error_schedules_reconnect_immediately` from #134 still passes.
- [x] `cargo test` — 63 lib tests + 1 doc test pass.
- [x] `cargo fmt --check` clean.

[SDK-2347]: https://launchdarkly.atlassian.net/browse/SDK-2347?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

---

> [!NOTE]
> **Medium Risk**
> Changes the SSE streaming state machine and retry backoff behavior, which can alter reconnect/termination semantics for consumers; scope is contained and covered by new tests.
> 
> **Overview**
> **Fixes reconnect disablement edge cases in the SSE client.** When `ReconnectOptions.reconnect` is `false`, parse errors and clean EOF now transition the stream to `StreamClosed` so the next poll returns `None` instead of scheduling a reconnect or continuing with a poisoned parser.
> 
> **Hardens retry backoff against server-supplied `retry: 0`.** `BackoffRetry::change_base_delay` now clamps the base delay to a 1ms minimum to avoid zero-delay reconnect loops, with a new unit test and additional stream-level tests covering the new close-on-error/EOF behavior.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit d33131a4e850003d8a44f30c0d0b2ad24519e6a0. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: eventsource-client/src/retry.rs [hunk 1]
```diff
@@ -15,6 +15,11 @@ pub(crate) trait RetryStrategy {
 
 const DEFAULT_RESET_RETRY_INTERVAL: Duration = Duration::from_secs(60);
 
+/// Floor applied to a server-supplied SSE `retry:` value. A server that
+/// sends `retry: 0` would otherwise collapse the backoff to zero and
+/// reconnect would become a tight loop.
+const MINIMUM_BASE_DELAY: Duration = Duration::from_millis(1);
```
Comment (kinyoklion) [inline eventsource-client/src/retry.rs:21]: We may want this larger.

**Review by keelerm84 — APPROVED** [review]
